panic: interface conversion: interface {} is string, not float64 (zoomeye)

Running from `dev | main`

```
./uncover -zoomeye 'title:"gogs"'
````

```console
94.103.206.187:443
110.40.191.14:443
110.42.188.156:443
110.42.193.145:443
128.199.39.72:8443
92.242.45.237:443
119.45.254.3:443
202.164.44.246:3000
panic: interface conversion: interface {} is string, not float64

goroutine 20 [running]:
github.com/projectdiscovery/uncover/uncover/agent/zoomeye.(*Agent).query(0x0?, {0x100bafdde?, 0x0?}, 0x0?, 0x0?, 0x0?)
	/Users/geekboy/Github/uncover/uncover/agent/zoomeye/zoomeye.go:104 +0x3c0
github.com/projectdiscovery/uncover/uncover/agent/zoomeye.(*Agent).Query.func1()
	/Users/geekboy/Github/uncover/uncover/agent/zoomeye/zoomeye.go:51 +0xd0
created by github.com/projectdiscovery/uncover/uncover/agent/zoomeye.(*Agent).Query
	/Users/geekboy/Github/uncover/uncover/agent/zoomeye/zoomeye.go:40 +0x134
```

--- a/uncover/agent/zoomeye/zoomeye.go
+++ b/uncover/agent/zoomeye/zoomeye.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"strconv"
 
 	"github.com/pkg/errors"
 
@@ -101,7 +102,10 @@ func (agent *Agent) query(URL string, session *uncover.Session, zoomeyeRequest *
 		}
 		if portinfo, ok := zoomeyeResult["portinfo"]; ok {
 			if port, ok := portinfo.(map[string]interface{}); ok {
-				result.Port = int(port["port"].(float64))
+				result.Port = convertPortFromValue(port["port"].(float64))
+				if result.Port == 0 {
+					continue
+				}
 				result.Host = port["hostname"].(string)
 				raw, _ := json.Marshal(zoomeyeResult)
 				result.Raw = raw
@@ -120,4 +124,16 @@ func (agent *Agent) query(URL string, session *uncover.Session, zoomeyeRequest *
 type ZoomEyeRequest struct {
 	Query string
 	Page  int
+}
+
+func convertPortFromValue(value interface{}) int {
+	switch v := value.(type) {
+	case float64:
+		return int(v)
+	case string:
+		parsed, _ := strconv.Atoi(v)
+		return parsed
+	default:
+		return 0
+	}
 }

